pkg/ruler/ruler.go

```diff
@@ -167,6 +173,10 @@ func (cfg *Config) Validate(limits validation.Limits, log log.Logger) error {
 		return errors.Wrap(err, "invalid ruler gRPC client config")
 	}
 
+	if _, err := url.Parse(cfg.FrontendAddress); err != nil {
+		return errInvalidFrontendAddress
+	}
```

Comment:
This check is unnecessary to me. If the frontend address is wrong, it should be captured at `promapi.NewClient`. Can we use that instead?